pkg/services/dns/dns.go

```diff
@@ -1,22 +1,66 @@
 package dns
 
 import (
-	"context"
 	"encoding/json"
 	"fmt"
 	"net"
 	"net/http"
+	"os"
 	"strings"
 	"sync"
+	"time"
 
 	"github.com/containers/gvisor-tap-vsock/pkg/types"
 	"github.com/miekg/dns"
 	log "github.com/sirupsen/logrus"
 )
 
 type dnsHandler struct {
-	zones     []types.Zone
-	zonesLock sync.RWMutex
+	zones      []types.Zone
+	zonesLock  sync.RWMutex
+	dnsClient  *dns.Client
+	nameserver string
+}
+
+func newDnsHandler(zones []types.Zone) *dnsHandler {
+
+	dnsClient, nameserver := readAndCreateClient()
+
+	return &dnsHandler{
+		zones:      zones,
+		dnsClient:  dnsClient,
+		nameserver: nameserver,
+	}
+
+}
+
+func readAndCreateClient() (*dns.Client, string) {
+	conf, err := dns.ClientConfigFromFile("/etc/resolv.conf")
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(2)
+	}
+	nameserver := conf.Servers[0]
```

Comment:
Fwiw, if there are multiple servers listed in /etc/resolv.conf, it's better if we keep the whole list, and use the additional ones as fallbacks if the first server does not answer. This can be refined later on.